### PR TITLE
bigquery: initial commit of snippets

### PR DESCRIPTION
**Pull request by rakyll**

### CODE REVIEW

File: bigquery/snippets/snippets_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	// [START bigquery_copy_table]
+	dataset := client.Dataset(datasetID)
+	copier := dataset.Table(dstID).CopierFrom(dataset.Table(srcID))
+	copier.WriteDisposition = bigquery.WriteTruncate
+	job, err := copier.Run(ctx)
+	if err != nil {
+		return err
+	}
+
+	for {
```
Comment (jba) [inline bigquery/snippets/snippets_test.go]: I just added Job.Wait, so this will get much simpler.

File: bigquery/snippets/snippets_test.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+
+	"cloud.google.com/go/bigquery"
+	"google.golang.org/api/iterator"
+)
+
+func TestAll(t *testing.T) {
```
Comment (broady) [inline bigquery/snippets/snippets_test.go]: Split into snippets.go and snippets_test.go

File: bigquery/snippets/snippets_test.go [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+	// [END bigquery_create_table]
+	return nil
+}
+
+func insertRows(client *bigquery.Client, datasetID, tableID string) error {
+	ctx := context.Background()
+	// [START bigquery_insert_stream]
+	u := client.Dataset(datasetID).Table(tableID).Uploader()
+	items := []*Item{
+		// Item implements the ValueSaver interface.
```
Comment (broady) [inline bigquery/snippets/snippets_test.go]: Add region tags around the type definition, move it next to the insertRows type.

I *think* the docs should pick up both region tags and put them together.

```
// [START bigquery_insert_stream]
type ...
// [END bigquery_insert_stream]
func insertRows...
// [START bigquery_insert_stream]
```

File: bigquery/snippets/snippets_test.go [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	Name  string
+	Count int
+}
+
+// Save implements the ValueSaver interface.
+func (i *Item) Save() (map[string]bigquery.Value, string, error) {
+	return map[string]bigquery.Value{
+		"Name":  i.Name,
+		"Count": i.Count,
+	}, "", nil
```
Comment (broady) [inline bigquery/snippets/snippets_test.go]: What's the string? Used named return values?

File: bigquery/snippets/snippets_test.go [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+	}
+	if err := u.Put(ctx, items); err != nil {
+		return err
+	}
+	// [ENDSTART bigquery_insert_stream]
+	return nil
+}
+
+func copyTable(client *bigquery.Client, datasetID, srcID, dstID string) error {
+	ctx := context.Background()
```
Comment (broady) [inline bigquery/snippets/snippets_test.go]: `const pollInterval` up here

File: bigquery/snippets/snippets_test.go (hunk 3 above)
Comment (rakyll) [inline bigquery/snippets/snippets_test.go]: Done

File: bigquery/snippets/snippets_test.go (hunk 5 above)
Comment (rakyll) [inline bigquery/snippets/snippets_test.go]: Done

File: bigquery/snippets/snippets_test.go (hunk 4 above)
Comment (rakyll) [inline bigquery/snippets/snippets_test.go]: This is the insertID from the ValueSaver interface to dedup.

"// If non-empty, BigQuery will use InsertID to de-duplicate insertions
  // of this row on a best-effort basis."

File: bigquery/snippets/snippets_test.go (hunk 2 above)
Comment (rakyll) [inline bigquery/snippets/snippets_test.go]: Done

**Comment by rakyll** [comment]
All done, waiting for the Job.Wait to be merged to remove the poller.

**Comment by jba** [comment]
Job.Wait CL has landed.

**Comment by rakyll** [comment]
Job.Wait is done, PTAL.

**Comment by broady** [comment]
```
bigquery/snippets/snippets.go:101: job.Wait undefined (type *"cloud.google.com/go/bigquery".Job has no field or method Wait)
```

**Pull request merged**